### PR TITLE
Add CSP ready to test live

### PR DESCRIPTION
**Pull request by TVke**

### CODE REVIEW

File: app/CustomCsp.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public function configure()
+    {
+        parent::configure();
+
+        $this
+            ->addDirective(Directive::BASE, 'self')
+            ->addDirective(Directive::STYLE, ['https://cloud.typography.com', 'https://spatie.be'])
+            ->addDirective(Directive::FONT, 'data:')
+            ->addDirective(Directive::SCRIPT, 'data:')
+            ->addDirective(Directive::CHILD, 'https://www.googletagmanager.com')
```
Comment (freekmurze) [inline app/CustomCsp.php]: We are using analytics on this site (through tag manager), could you set up the necessary directives for this as well? Take a look at murze.be for an example.

File: resources/views/layouts/app.blade.php [hunk 2]
```diff
@@ -17,7 +17,7 @@
     <link href="{{ mix('css/app.css') }}" rel="stylesheet">
     <link rel="stylesheet" type="text/css" href="https://cloud.typography.com/6194432/608542/css/fonts.css" />
 
-    <script src="{{ mix('js/app.js') }}" defer></script>
+    <script @nonce src="{{ mix('js/app.js') }}" defer></script>
```
Comment (freekmurze) [inline resources/views/layouts/app.blade.php]: nice!

File: tests/CreatesApplication.php [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 namespace Tests;
 
-use Illuminate\Support\Facades\Hash;
 use Illuminate\Contracts\Console\Kernel;
+use Illuminate\Support\Facades\Hash;
```
Comment (freekmurze) [inline tests/CreatesApplication.php]: Why this change?

File: app/CustomCsp.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App;
+
+use Spatie\Csp\Directive;
+use Spatie\Csp\Policies\Basic;
+
+class CustomCsp extends Basic
```
Comment (freekmurze) [inline app/CustomCsp.php]: Rename this class to `Policy` and put it in `app\Services\Csp`.

File: content/code-style/css.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # CSS Style Guide
 
-- [Preprocessing](#preprosessing)
+- [Preprocessing](#preprocessing)
```
Comment (freekmurze) [inline content/code-style/css.md]: Nice catches!

If you spot some more, open up a separate PR for typos, so I can merge that it independently of the csp changes.

Leave these as they are for now.

Comment (TVke) [reply content/code-style/css.md]: I will definitely think about it next time.

**Comment by freekmurze** [comment]
Thanks, do follow up on this and let me know when we can use the policy for real (without report only)
